src/language-json/languages.evaluate.js

```diff
@@ -15,10 +15,9 @@ const languages = [
     vscodeLanguageIds: ["json"],
     extensions: data.extensions.filter((extension) => extension !== ".jsonl"),
   })),
-  createLanguage(linguistLanguages["JSON with Comments"], (data) => ({
-    parsers: ["json"],
+  createLanguage(linguistLanguages["JSON with Comments"], () => ({
+    parsers: ["jsonc"],
     vscodeLanguageIds: ["jsonc"],
-    filenames: [...data.filenames, ".eslintrc"],
```

Comment:
`.eslintrc` is not listed in https://eslint.org/docs/latest/use/configure/configuration-files#configuration-file-formats, and `.eslintrc.json` already in `linguistLanguages`.